src/setups/extrude_geometry.jl

```diff
@@ -0,0 +1,212 @@
+    edge1 = plane_points[2] - plane_points[1]
+    edge2 = plane_points[3] - plane_points[1]
+
+    # Check if the points are collinear
+    if norm(cross(edge1, edge2)) == 0
+        throw(ArgumentError("the points must not be collinear"))
+    end
+
+    # Determine the number of points along each edge
+    num_points_edge1 = ceil(Int, norm(edge1) / particle_spacing)
+    num_points_edge2 = ceil(Int, norm(edge2) / particle_spacing)
+
+    coords = zeros(3, (num_points_edge1 + 1) * (num_points_edge2 + 1))
+
+    index = 1
+    for i in 0:num_points_edge1
+        for j in 0:num_points_edge2
+            point_on_plane = plane_points[1] + (i / num_points_edge1) * edge1 +
+                             (j / num_points_edge2) * edge2
+            coords[:, index] = point_on_plane
+            index += 1
+        end
+    end
+
+    particle_spacing_new = min(norm(edge1 / num_points_edge1),
+                               norm(edge2 / num_points_edge2))
+
+    return coords, particle_spacing_new
+end
+
+function consider_particle_placement(geometry::Union{AbstractMatrix, InitialCondition},
+                                     direction, particle_spacing, tlsph)
+    return geometry
+end
+
+function consider_particle_placement(plane_points, direction, particle_spacing, tlsph)
+    consider_particle_placement(tuple(plane_points...), direction, particle_spacing, tlsph)
+end
+
+function consider_particle_placement(plane_points::NTuple{2}, direction, particle_spacing,
+                                     tlsph)
+    # With TLSPH, particles need to be AT the min coordinates and not half a particle
+    # spacing away from it.
+    (tlsph) && (return plane_points)
+
+    plane_point1 = copy(plane_points[1])
+    plane_point2 = copy(plane_points[2])
+
+    # Vectors shifting the points in the corresponding direction
+    dir1 = 0.5 * particle_spacing * direction
+    dir2 = 0.5 * particle_spacing * normalize(plane_point2 - plane_point1)
+
+    plane_point1 .+= dir1 + dir2
+    plane_point2 .+= dir1 - dir2
+
+    return (plane_point1, plane_point2)
+end
+
+function consider_particle_placement(plane_points::NTuple{3}, direction, particle_spacing,
+                                     tlsph)
+    # With TLSPH, particles need to be AT the min coordinates and not half a particle
+    # spacing away from it.
+    (tlsph) && (return plane_points)
+
+    plane_point1 = copy(plane_points[1])
+    plane_point2 = copy(plane_points[2])
+    plane_point3 = copy(plane_points[3])
+
+    edge1 = normalize(plane_point2 - plane_point1)
+    edge2 = normalize(plane_point3 - plane_point1)
+
+    # Vectors shifting the points in the corresponding direction
+    dir1 = 0.5 * particle_spacing * direction
+    dir2 = 0.5 * particle_spacing * edge1
+    dir3 = 0.5 * particle_spacing * edge2
+
+    plane_point1 .+= dir1 + dir2 + dir3
+    plane_point2 .+= dir1 - dir2 + dir3
+    plane_point3 .+= dir1 + dir2 - dir3
+
+    return (plane_point1, plane_point2, plane_point3)
+end
```

Comment:
```suggestion
function adjust_plane_points(plane_points::NTuple{N, <:Any}, direction, particle_spacing) where N
    new_points = ()

    dir_shift = 0.5 * particle_spacing * direction

    for (i, point) in enumerate(plane_points)
        dir = dir_shift

        for (j, other_point) in enumerate(plane_points)
            if i != j
                dir += 0.5 * particle_spacing * normalize(other_point - point)
            end
        end

        new_point = copy(point) .+= dir
        new_points = (new_points..., new_point)
    end

    return new_points
end

function consider_particle_placement(plane_points::NTuple{2}, direction, particle_spacing, tlsph)
    (tlsph) && (return plane_points)
    return adjust_plane_points(plane_points, direction, particle_spacing)
end

function consider_particle_placement(plane_points::NTuple{3}, direction, particle_spacing, tlsph)
    (tlsph) && (return plane_points)
    return adjust_plane_points(plane_points, direction, particle_spacing)
end
```
Something like this (untested)